libs-scala/resources/src/main/scala/com/digitalasset/resources/ProgramResource.scala

```diff
@@ -20,7 +21,12 @@ final class ProgramResource[Context: HasExecutionContext, T](
 ) {
   private val logger = ContextualizedLogger.get(getClass)
 
-  private val executorService = Executors.newCachedThreadPool()
+  private val executorService = {
+    val counter = new AtomicLong(0L)
+    Executors.newCachedThreadPool((runnable: Runnable) =>
+      new Thread(runnable, s"program-resource-pool-${counter.incrementAndGet()}")
+    )
```

Comment:
Why not use a `ThreadFactoryBuilder` here?

Reply:
Because I would have to add the dependency to google-commons to the `libs-scala/resources` library. And considering that this is a small utility library, I'd rather keep the dependency surface small. But maybe this actually asks for extracting out [NamedThreadFactory](https://github.com/digital-asset/daml/blob/main/runtime-components/non-repudiation-resources/src/main/scala/com/daml/nonrepudiation/resources/HikariTransactorResourceOwner.scala#L30) into a common place.

Reply:
Perhaps leave it for now and let's see if we use this a second time. Didn't realize the builder was part of Guava.